plugins/core-plugin/src/main/java/com/google/cloud/teleport/plugin/PythonDockerfileGenerator.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.plugin;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.logging.Logger;
+
+/** Utility class that generates a simple Dockerfile for Python templates. */
+public final class PythonDockerfileGenerator {
+
+  private static final Logger LOG = Logger.getLogger(PythonDockerfileGenerator.class.getName());
+
+  private PythonDockerfileGenerator() {}
+
+  public static void generateDockerfile(
+      String basePythonContainerImage, String containerName, File targetDirectory)
+      throws IOException {
+    StringBuilder stringBuilder = new StringBuilder();
+    stringBuilder.append(
+        "FROM " + basePythonContainerImage + System.lineSeparator() + System.lineSeparator());
+    stringBuilder.append("ARG WORKDIR=/template" + System.lineSeparator());
+    stringBuilder.append("RUN mkdir -p ${WORKDIR}" + System.lineSeparator());
+    stringBuilder.append("COPY " + containerName + "/main.py ${WORKDIR}" + System.lineSeparator());
+    stringBuilder.append(
+        "COPY " + containerName + "/requirements.txt ${WORKDIR}" + System.lineSeparator());
+    stringBuilder.append("WORKDIR ${WORKDIR}" + System.lineSeparator() + System.lineSeparator());
+    stringBuilder.append(
+        "ENV FLEX_TEMPLATE_PYTHON_REQUIREMENTS_FILE=requirements.txt" + System.lineSeparator());
```

Comment:
It's good to have a separate generator. I think we could keep it more maintainable having the Dockerfile separately, and using something like Freemarker to replace the variables.

Take a look at TemplateDocsGenerator / README-template.md. We could likely have "Dockerfile-template" and use `${}` to interpolate content

Reply:
done